addons/hr_payroll/models/hr_benefit.py

```diff
@@ -133,34 +123,53 @@ def compute_conflicts_leaves_to_approve(self):
             })
         return bool(conflicts)
 
-    def safe_duplicate_create(self, vals):
+    def _safe_duplicate_create(self, vals_list):
         """
-        Create benefit but silently abord if it already exists in the database (according to the _unique constraints).
-        @return: new record if it didn't exist, empty recordset otherwise.
+        Create benefit but silently abord if it already exists in the database.
+        @return: new record id if it didn't exist.
         """
-        try:
-            with mute_logger('odoo.sql_db'), self.env.cr.savepoint():
-                # Any database call will be run inside a postgresql savepoint which is
-                # rolledback if an exception occurs allowing to make subsequent calls in the transaction.
-                # (Otherwise InternalError is raised by any subsequent database operations)
-                return self.create(vals)
-        except IntegrityError as err:
-            if 'hr_benefit__unique' not in err.pgerror:
-                raise err
-            return self.env['hr.benefit']
+        query = """
+                INSERT INTO hr_benefit (
+                    name,
+                    date_start,
+                    date_stop,
+                    duration,
+                    benefit_type_id,
+                    display_warning,
+                    employee_id,
+                    state,
+                    leave_id,
+                    active)
+                VALUES {}
+                ON CONFLICT DO NOTHING
+                RETURNING id""".format(','.join(['%s'] * len(vals_list)))
+        params = [(
+            vals['name'],
+            vals['date_start'],
+            vals['date_stop'],
+            self._get_duration(vals['date_start'], vals['date_stop']),
+            vals.get('benefit_type_id', False) or None,
+            not bool(vals['benefit_type_id']),
+            vals['employee_id'],
+            'confirmed',
+            vals.get('leave_id', False) or None,
+            True) for vals in vals_list]
+        self.env.cr.execute(query, params)
+        result = self.env.cr.fetchone()
+        return result[0] if result else False
```

Comment:
Why not `self.browse(...)` ?

Reply:
It would probably allow to keep `def _compute_conflicts_leaves_to_approve` as `@api.multi`

Reply:
Yep, but as this method only make self.ids inside it, it's to avoid browsing all the benefits that have been created.